server/world/viewer.go

```diff
@@ -25,9 +25,10 @@ type Viewer interface {
 	// ViewEntityTeleport views the teleportation of an entity. The entity is immediately moved to a different
 	// target position.
 	ViewEntityTeleport(e Entity, position mgl64.Vec3)
-	// ViewChunk views the chunk passed at a particular position. It is called for every chunk loaded using
-	// the world.Loader.
-	ViewChunk(pos ChunkPos, c *chunk.Chunk, blockNBT map[cube.Pos]Block)
+	// ViewSubChunks views the sub-chunks requested in the offsets passed.
+	ViewSubChunks(center SubChunkPos, offsets [][3]int8)
+	// ViewSkeletonChunk views the skeleton chunk passed. It is called before the viewer requests sub-chunks.
+	ViewSkeletonChunk(pos ChunkPos, c *chunk.Chunk)
```

Comment:
I'd name this `ViewChunk`, the fact that it is sent as a skeleton chunk is an implementation detail specific to a client Viewer.

Reply:
Should I also re-add the block NBT parameter, ignoring it in the session implementation?

Reply:
Nah you don't have to do that.